### PR TITLE
feat: candidate Feature File for Task List

### PR DESCRIPTION
**Pull request by karunak21**
## Ticket Number
UCD-766

## Description of change
Candidate Feature file for Your Details section in Task List page

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [ ] I have merged commits to avoid fixing things in this PR
- [ X] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Comment by karunak21** [comment]
We would like to reuse existing Gherkin sentences but those sentences need to be fixed to allow reuse. 
If we are happy with the feature sentences then @danWatkinson  has offered help to work on the existing sentences.

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        # correct this. update who=are=you feature to accept "yes"/"no" or "is"/"isnt"
+        And the user states that they <original appelant> the original appellant
+        # correct this. update who-are-you feature to care about the original apelant name - this sentence should appear in that feature somewhere?
+        And the user provides <on behalf of> as the name of the original appellant
+        Then the user should see that Your Details section is <status>
+        
+        Examples:
+            | name                | email            | original apellant | on behalf of    | Status        |
+            | ""                  | ""               | "doesnt matter"   | "doesnt matter" | "Not Started" |
+            | ""                  | "good@email.com" | "doesnt matter"   | "doesnt matter" | "Not Started" |
```
Comment (frank-dodion) [inline e2e-tests/cypress/integration/appeal-submission-tasklist.feature:16]: Will it be possible to get into this state where there is no name but there is an email? I think if this is submitted it should fail validation and prevent any update.

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@wip
+Feature: Your Details status
+
+    Scenario Outline: The status of the Your Details section is accurate
+        # correct this, and correct your-details feature; 'valid' should not feature here; we want to be able to re-use for invalid cases too
```
Comment (frank-dodion) [inline e2e-tests/cypress/integration/appeal-submission-tasklist.feature:5]: I am not sure we should have any comments in the feature file.

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature (hunk 1 above)
Comment (danWatkinson) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:16]: that's the situation we're trying to communicate here - so if we give no name + an email we should fail validation so the status should be 'not started'.
totally aware there may be nicer ways to word things but i was trying to show that in a situation where we've given something invalid:  it doesn't matter what we do for the other fields (maybe "" is clearer there?)

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature (hunk 2 above)
Comment (danWatkinson) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:5]: we 100% shouldn't
however -  we're proposing that we'd like to re-use some existing sentences but the existing sentences aren't quite the right shape... if we're happy with the proposed syntax I'll pick this up and go re-work the existing sentences+glue so we get the re-use.. i just can't be bothered to do that until we know the new structure is 👍

Comment (frank-dodion) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:5]: Np. thanks

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature (hunk 1 above)
Comment (frank-dodion) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:16]: Ok.

**Review by frank-dodion — APPROVED** [review]

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+        Examples:
+            | name        | email            | Status        |
+            | ""          | ""               | "Not Started" |
+            | ""          | "good@email.com" | "Not Started" |
+            | "Good Name" | ""               | "Not Started" |
+            | "Good Name" | "good@email.com" | "Completed"   |
+
+
+      Scenario Outline: The status of the Your Details section is accurate
```
Comment (frank-dodion) [inline e2e-tests/cypress/integration/appeal-submission-tasklist.feature:18]: I think we could use a different wording in the title to show it is different from the scenario above i.e. in this case "... when the user is not the original applicant"

**Review by frank-dodion — APPROVED** [review]
We could refactor scenario title later but getting this in now will allow work on the steps to continue.

File: e2e-tests/cypress/integration/appeal-submission-tasklist.feature (hunk 3 above)
Comment (karunak21) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:18]: Will offer tidy up in glue code

Comment (karunak21) [reply e2e-tests/cypress/integration/appeal-submission-tasklist.feature:18]: This will be addressed while implementing the glue code

**Pull request merged**